components/navigation/BlogPostItem.tsx

```diff
@@ -0,0 +1,137 @@
+import moment from 'moment';
+import Link from 'next/link';
+import type { Ref } from 'react';
+import { forwardRef } from 'react';
+import TextTruncate from 'react-text-truncate';
+
+import { HeadingLevel, HeadingTypeStyle } from '@/types/typography/Heading';
+import { ParagraphTypeStyle } from '@/types/typography/Paragraph';
+
+import AuthorAvatars from '../AuthorAvatars';
+import Heading from '../typography/Heading';
+import Paragraph from '../typography/Paragraph';
+
+interface Author {
+  name: string;
+  photo: string;
+  link?: string;
+}
+
+interface Post {
+  type: string;
+  alt?: string;
+  slug: string;
+  cover: string;
+  title: string;
+  excerpt: string;
+  authors: Author[];
+  date: string;
+  readingTime: number;
+}
```

Comment:
Are you sure to use the `Post` type like this? We have already defined there types in `types` folder.

Reply:
Hmm yes, that one is the preferred way.